### PR TITLE
add client-side error handling

### PR DESCRIPTION
**Pull request by janbehrens**

### CODE REVIEW

File: webapp/views/launch/script.js [hunk 1]
```diff
@@ -530,28 +533,29 @@ function saveDrawing(res) {
...
-  get('/output', {}, function (res) {
-    const baseOption = "<option selected value=''> - </option>"
-    const options = res.list.reduce((str, file) => str + `<option value="${file}">${file}</option>`, '')
-    $('#results-select').html(baseOption + options)
-  })
+  get('/output', {}, (res) => new Promise(() => {
+    const baseOption = "<option selected value=''> - </option>";
+    const options = res.list.reduce((str, file) => str + `<option value="${file}">${file}</option>`, '');
+    $('#results-select').html(baseOption + options);
+  }));
```
Comment (JowieXiang) [inline webapp/views/launch/script.js:540]: Shouldn't this Promise have a `resolve` as well?

File: webapp/views/launch/script.js [hunk 2]
```diff
@@ -530,28 +533,29 @@ function saveDrawing(res) {
...
-    $('#table-header').text(`Table description for ${tableObj.table}`)
-    $('#table-description').text(tableObj.description)
-    $('#column-description').html(tableElement('table table-bordered', cObj))
-    $('#table-attributes-modal').show()
-  })
+    $('#table-header').text(`Table description for ${tableObj.table}`);
+    $('#table-description').text(tableObj.description);
+    $('#column-description').html(tableElement('table table-bordered', cObj));
+    $('#table-attributes-modal').show();
+  }));
```
Comment (JowieXiang) [inline webapp/views/launch/script.js:558]: `Resolve`?

**Review by JowieXiang — APPROVED** [review]
Nice new function!
The `'resolve`' thing is just a coding style problem I guess.

File: webapp/views/launch/script.js (hunk 1 above)
Comment (janbehrens) [reply webapp/views/launch/script.js:540]: I works this way, too, because it resolves by default, but I agree, resolve should be called for clarity.

**Pull request merged**